### DISCUSSION TITLE
empty sync_list file maybe causing skip-dir to be ignored.

### DISCUSSION THREAD

**Discussion by mekaniker8** [bpozdena/OneDriveGUI#206] [Q&A]
Hi,

new onedrivegui user here. I have been running into some issues during my initial tests with onedrivegui. 
I am still working on reproducing this step by step, but my observation is that the gui may be creating an empty sync_list file which causes the skip-dir variables to be rejected. I will post more info once I know more.

**Comment by abraunegg** [comment]
> empty sync_list file maybe causing skip-dir to be ignored

If this is the case - this is not a GUI issue .. but a 'onedrive' client issue .......

**Comment by bpozdena** [comment]
@mekaniker8 , yes, it's true that the GUI creates an empty `sync_list`  file  when it contains no configuration and the settings are saved.  I can change this behavior, but I've never seen this to cause any issues with the onedrive client.

Have you been able to collect more details about the issue you experienced with skip_dir?

**Comment by abraunegg** [comment]
@bpozdena 
My advice here is do not create an empty 'sync_list' file .. it should only be created when the user enters in a 'sync_list' configuration , and if that configuration is removed (zero entries), it the file should be removed.

It appears that this issue might also be the cause of this: https://github.com/abraunegg/onedrive/discussions/2995

In either case, an 'empty' sync_list file should not be causing 'skip_dir' entries to not be matched

**Comment by abraunegg** [comment]
@bpozdena 
The creation and importation of an empty 'sync_list' file is fixed via https://github.com/abraunegg/onedrive/pull/2996

Now this needs to be validated against 'skip_dir' rules not being applied.

**Comment by mekaniker8** [comment]
@bpozdena yes, the issue is rather clear to me now. 
here is how i can reproduce it. 

1. before setting up onedrive gui, i create a config file under ~/.config/onedrive/ which only includes skip_dir rules for multiple folders.
2. i open onedrive gui for first time, import the existing config file and finish setup. the skip_dir folders are correctly visible on the gui. 
3. everything works fine. 
4. i change an option on the gui such as enable auto-sync at startup, this creates the sync_list file and requires me to resync which i believe it shouldn't do for changing such an option but i assume the reason is that a sync_list has been created now.
5. the resync does not respect the skip_dir folders and starts syncing all folders on my onedrive. 

upon which, i stop the sync, delete the profile from gui, then delete the unwanted folders, and delete the sync_list file and then resync with onedrive without issues.  now i can start gui again and import the config again, this time not touching the auto-sync at startup option in the gui, just leaving everything alone basically and things work well.. 

hope that helps.

**Comment by abraunegg** [comment]
@mekaniker8 
There are a number of things going on here:

> 4. i change an option on the gui such as enable auto-sync at startup, this creates the sync_list file ...

This is a GUI bug. The GUI should not be creating an empty 'sync_list' file if no entries to actually add|modify|delete. This behavior should be corrected. If there are no entries, with this option enabled, this file should not exist in the user configuration.

> ... this creates the sync_list file and requires me to resync which i believe it shouldn't do for changing such an option but i assume the reason is that a sync_list has been created now.

The 'sync_list' use - creation, modification, deletion triggers a `--resync` requirement. Please read:
* https://github.com/abraunegg/onedrive/blob/master/docs/usage.md#using-client-side-filtering-rules-to-determine-what-should-be-synced-with-microsoft-onedrive
* https://github.com/abraunegg/onedrive/blob/master/docs/usage.md#performing-a-selective-synchronisation-via-sync_list-file

This is *not* a bug, and this behaviour will not be corrected or changed.

> 5. the resync does not respect the skip_dir folders and starts syncing all folders on my onedrive.

This is an application bug , not a GUI bug. This bug has already been remediated:
* https://github.com/abraunegg/onedrive/pull/2996
* https://github.com/abraunegg/onedrive/pull/2998
